src/tribler-core/tribler_core/utilities/torrent_utils.py

```diff
@@ -17,34 +17,27 @@ def commonprefix(paths_list):
     return sorted(base_set, reverse=True)[0]
 
 
-def create_torrent_file(file_path_list, params, torrent_filepath=None):
-    fs = lt.file_storage()
-
-    # filter all non-files
-    file_path_list_filtered = []
-    for path in file_path_list:
-        path = path_util.Path(path)
+def _existing_files(path_list):
+    for path in path_list:
+        path = Path(path)
         if not path.exists():
             raise OSError(f'Path does not exist: {path}')
         elif path.is_file():
-            file_path_list_filtered.append(path)
+            yield path
 
-    # get the directory where these files are in. If there are multiple files, take the common directory they are in
-    if len(file_path_list_filtered) == 1:
-        base_path = path_util.split(file_path_list_filtered[0])[0]
-    else:
-        base_path = path_util.abspath(commonprefix(file_path_list_filtered))
 
-    # the base_dir directory is the parent directory of the base_path and is passed to the set_piece_hash method
+def create_torrent_file(file_path_list, params, torrent_filepath=None):
+    fs = lt.file_storage()
+
+    # filter all non-files
+    path_list = list(_existing_files(file_path_list))
 
-    if len(file_path_list_filtered) == 1:
-        filename = path_util.basename(file_path_list_filtered[0])
-        fs.add_file(filename, path_util.getsize(file_path_list_filtered[0]))
-    else:
-        for full_file_path in file_path_list_filtered:
-            #FIXME: there should be a better, cleaner way to define this
-            filename = path_util.join(*full_file_path.parts[len(base_path.parent.parts):])
-            fs.add_file(str(filename), path_util.getsize(full_file_path))
+    # get the directory where these files are in. If there are multiple files, take the common directory they are in
+    base_path = commonprefix(path_list).parent if len(path_list) > 1 else path_list[0].parent
+    base_path = base_path.absolute()
+    for path in path_list:
+        relative = path.relative_to(base_path)
+        fs.add_file(str(relative), path.size())
```

Comment:
Right now I keep the original logic.

If there is only one file, then the values will be the following:

```
path_list = ['/user/tmp/1.txt']
name='1.txt'
base_path='/user/tmp'
```

If two or more files, then

```python
path_list = ['/user/tmp/1.txt', '/user/tmp/2.txt']
names = ['tmp/1.txt', 'tmp/2.txt']
base_path='/user'
```

My question here: "Should we actually keep the origin logic?"
Or, in the case of two and more files, it is better to have:

```python
path_list = ['/user/tmp/1.txt', '/user/tmp/2.txt']
names = ['1.txt', '2.txt']
base_path='/user/tmp'
```